spark/v3.5/spark-extensions/src/main/scala/org/apache/spark/sql/catalyst/parser/extensions/IcebergSqlExtensionsAstBuilder.scala

```diff
@@ -227,8 +227,10 @@ class IcebergSqlExtensionsAstBuilder(delegate: ParserInterface) extends IcebergS
 
     val distributionMode = if (distributionSpec != null) {
       DistributionMode.HASH
-    } else if (orderingSpec.UNORDERED != null || orderingSpec.LOCALLY != null) {
+    } else if (orderingSpec.UNORDERED != null) {
       DistributionMode.NONE
+    } else if (orderingSpec.LOCALLY() != null) {
+      null
```

Comment:
This is a bit confusing to me since I don't think we have it well defined in SetWriteDistributionAndOrdering that null means don't change. I'm also generally against using null if possible in Scala.

I think this may require refactoring the SetWriteDistributionAndOrdering to be a little more flexible? It could take Option(DistributionMode), Option(SortOrder)?